deps/rabbit/src/rabbit_incoming_message_interceptor.erl

```diff
@@ -0,0 +1,38 @@
+-module(rabbit_incoming_message_interceptor).
```

Comment:
i don't think we need to distinguish incoming and outgoing interceptors are the module / behaviour level.
We should have a single behaviour `rabbit_message_interceptor` with a single `intercept` function that takes the message container, the protocol context map (see another comment) and the config.

Instead they would be distinguined by which configuration they are added to. It is fine if this PR only implements incoming interceptors.